src/inference/src/dev/threading/cpu_streams_executor.cpp

```diff
@@ -335,27 +334,75 @@ struct CPUStreamsExecutor::Impl {
     // will be counted by thread_local t_stream_count_map.
     // when the customer's thread is destoryed, the stream's count will became 1,
     // Call local() will reuse one of them, and release others.
+    // it's only a workaround for ticket CVS-111490, please be carefully when need to modify
+    // CustomeThreadLocal::local(), especially like operations that will affect the count of
+    // CustomThreadLocal::ThreadId
     class CustomThreadLocal : public ThreadLocal<std::shared_ptr<Stream>> {
+        class ThreadId {
+        public:
+            explicit ThreadId(std::thread::id& id)
+                : _id_ptr(new std::thread::id(id)),
+                  _count_ptr(new std::atomic_int(1)) {}
+            ~ThreadId() {
+                if (_count_ptr->fetch_sub(1) == 1) {
+                    delete _id_ptr;
+                    delete _count_ptr;
+                }
+            }
+            ThreadId* copy() {
+                ThreadId* new_ptr = new ThreadId(*this);
+                auto pre_valule = new_ptr->_count_ptr->fetch_add(1);
+                OPENVINO_ASSERT(pre_valule == 1, "this value must be 1, please check code CustomThreadLocal::local()");
+                return new_ptr;
+            }
+            std::thread::id& get_id() {
+                return *_id_ptr;
+            }
+            int count() {
+                return *_count_ptr;
+            }
+
+        private:
+            // disable all copy and move semantics, user only can use copy()
+            // to create a new instance with a shared count num;
+            ThreadId(ThreadId const&) = default;
+            ThreadId(ThreadId&&) = delete;
+            ThreadId& operator=(ThreadId const&) = delete;
+            ThreadId& operator=(ThreadId&&) = delete;
+            std::thread::id* _id_ptr;
+            std::atomic_int* _count_ptr;
```

Comment:
It is simplier to use `shared_ptr` so that to avoid a situation of dereferencing already deleted object.